api/src/main/java/org/pmiops/workbench/cohortbuilder/SearchGroupItemQueryBuilder.java

```diff
@@ -0,0 +1,578 @@
+          + "and not exists (\n"
+          + "SELECT 'x' FROM `${projectId}.${dataSetId}.death` d\n"
+          + "where d.person_id = p.person_id)\n";
+  private static final String RACE_SQL = "p.race_concept_id in unnest(%s)\n";
+  private static final String GEN_SQL = "p.gender_concept_id in unnest(%s)\n";
+  private static final String ETH_SQL = "p.ethnicity_concept_id in unnest(%s)\n";
+  private static final String DEC_SQL =
+      "exists (\n"
+          + "SELECT 'x' FROM `${projectId}.${dataSetId}.death` d\n"
+          + "where d.person_id = p.person_id)\n";
+
+  /** Build the inner most sql using search parameters, modifiers and attributes. */
+  public static void buildQuery(
+      Map<SearchParameter, Set<Long>> criteriaLookup,
+      Map<String, QueryParameterValue> queryParams,
+      List<String> queryParts,
+      SearchGroup searchGroup,
+      boolean isEnableListSearch) {
+    if (searchGroup.getTemporal()) {
+      // build the outer temporal sql statement
+      String query =
+          buildOuterTemporalQuery(criteriaLookup, queryParams, searchGroup, isEnableListSearch);
+      queryParts.add(query);
+    } else {
+      for (SearchGroupItem searchGroupItem : searchGroup.getItems()) {
+        // build regular sql statement
+        String query;
+        if (isEnableListSearch) {
+          query =
+              buildBaseQuery(
+                  criteriaLookup, queryParams, searchGroupItem, searchGroup.getMention());
+        } else {
+          // TODO:Remove when new search is finished - freemabd
+          query =
+              QueryBuilderFactory.getQueryBuilder(FactoryKey.getType(searchGroupItem.getType()))
+                  .buildQuery(queryParams, searchGroupItem, searchGroup.getMention());
+        }
+        queryParts.add(query);
+      }
+    }
+  }
+
+  /** Build the inner most sql */
+  private static String buildBaseQuery(
+      Map<SearchParameter, Set<Long>> criteriaLookup,
+      Map<String, QueryParameterValue> queryParams,
+      SearchGroupItem searchGroupItem,
+      TemporalMention mention) {
+    Set<Long> childConceptIds = new HashSet<>();
+    boolean isStandard = false;
+    List<String> queryParts = new ArrayList<>();
+
+    // When building sql for demographics - we query against the person table
+    if (DomainType.PERSON.toString().equals(searchGroupItem.getType())) {
+      return buildDemoSql(queryParams, searchGroupItem);
+    }
+    // Otherwise build sql against flat denormalized search table
+    for (SearchParameter param : searchGroupItem.getSearchParameters()) {
+      if (param.getAttributes().isEmpty()) {
+        childConceptIds.addAll(childConceptIds(criteriaLookup, ImmutableList.of(param)));
+        // make sure we only add the concept ids sql template once
+        if (!queryParts.contains(CONCEPT_ID_IN)) {
+          queryParts.add(CONCEPT_ID_IN);
+        }
+      } else {
+        StringBuilder bpSql = new StringBuilder(BP_SQL);
```

Comment:
I'm a bit lost here - why do we go straight into a blood pressure query? At this point we just have a searchGroupItem with attributes listed

Reply:
So `attribute.conceptId` - the **only** use case for this is blood pressure? If so, possibly add this to the comment at L180

Reply:
Yes that is true. Comment added.